### PR TITLE
Geocoder Cleanup

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
The OTP Geocoder change invites some room for improvement.

This PR addresses some crash conditions

### CODE REVIEW

File: __tests__/util.test.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ const HERE_RESPONSE_BUS =
   require('./json-mocks/here-response-bus.json') as FeatureCollection
 
 describe('arePointsEqual', () => {
+  it('should handle null inputs', () => {
+    // @ts-expect-error we are doing incorrect things
+    expect(arePointsRoughlyEqual([null, null], null)).toBe(false)
+    // @ts-expect-error we are doing incorrect things
+    expect(arePointsRoughlyEqual(null, [null, null])).toBe(false)
+    // @ts-expect-error we are doing incorrect things
```
Comment (daniel-heppner-ibigroup) [inline __tests__/util.test.ts:29]: yeah....

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
Not sure we should be working around the TypeScript rules like this. What's the problem being solved?

File: utils.ts [hunk 2]
```diff
@@ -128,14 +128,16 @@ export const fetchPelias = async (
  */
 export const arePointsRoughlyEqual = (a: Position, b: Position): boolean => {
   // 4 decimal places is approximately 10 meters, which is acceptable error
-  const aRounded = a.map((point: number): number =>
-    parseFloat(point.toFixed(4))
+  const aRounded = a?.map((point: number): number =>
```
Comment (daniel-heppner-ibigroup) [inline utils.ts:131]: What problem are we solving with the optional chaining? It shouldn't be necessary if the TypeScript types on the arguments are correct. I'm guessing that we're getting an API response back somewhere that doesn't always conform to the shape we've defined with TypeScript. Perhaps we should update that type so TypeScript can correctly show us where we're using the API response incorrectly. The answer in this function might be just to make the arguments nullable.

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
oops, I did not submit this comment with my first review

File: __tests__/util.test.ts (hunk 1 above)
Comment (miles-grant-ibigroup) [reply __tests__/util.test.ts:29]: It's for a test, so this is supposed to happen

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Perhaps we can take another look at the types in the future.

**Review by amy-corson-ibigroup — APPROVED** [review]

**Pull request merged**